### ISSUE TITLE
LINE、randomwalk

### ISSUE THREAD

**Issue by ivankxt** [alibaba/euler#108]
背景：10台机器，每台机器配置为64CPU、251G内存，数据规模9千万个点，20亿边，运行LINE或者RandomWalk算法模型，启动1ps + 20worker进行分布式训练，

执行如何命令：
`--data_dir hdfs://path/data
--euler_zk_addr 100.70.40.40@2181
--euler_zk_path /euler 
--max_id 90170696
--feature_idx 0 
--feature_dim 2
--model line
--model_dir /path/model_line_128_512
--mode train
--learning_rate 0.00005
--num_epochs 1
--dim 128
`
（1）当设置dim = 128时，load完数据后，加载模型过程中，抛出内存不足，运行失败。所以，想问一下像LINE这样的模型为什么这么耗内存？是不是执行命令不对？

（2）当设置dim = 8时，程序可以执行，但是PS进程占用的cpu比较严重，而对应的worker占用cpu不到1%，即所有压力落在的PS上，如下图所示：
![image](https://user-images.githubusercontent.com/42996988/58476877-aa0aa000-8184-11e9-8e48-20bb6ab19b56.png)
从日志打印看出，运行耗时严重：
![image](https://user-images.githubusercontent.com/42996988/58476965-f0f89580-8184-11e9-90a7-a511f460133d.png)

@yangsiran 
作为对比，同样的数据运行graphsage算法模型时（dim = 256），一个epoch运行耗时只需要50分钟；所以，对于euler上提供的LINE、RandomWalk等算法模型，基本上无法进行大规模数据的训练，请问，这个问题如何解决? 谢谢!

**Comment by yangsiran** [comment]
看起来可以增加PS的数目，并对Embeding layer里面的variable进行partition。

**Comment by ivankxt** [comment]
> 看起来可以增加PS的数目，并对Embeding layer里面的variable进行partition。

这个“Embeding layer里面的variable进行partition”具体怎么做呢？我们之前直接增加PS的数量，但是压力还是只落在主PS上，期望指导指导哈~

**Comment by ShangJP** [comment]
@ivankxt 遇到同样问题，您现在这个问题解决了吗，还有一个问题，就是在分布式训练line时，loss会边的异常的大，这个问题您清楚什么原因么

**Comment by alinamimi** [comment]
#176
